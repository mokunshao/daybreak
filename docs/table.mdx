---
title: Table
---

import {useState} from 'react'
import { Playground, Props } from 'docz'
import Table from '../src/table/table.tsx'

# Table

```javascript
import { Table } from 'daybreak';
```

## Standalone

<Playground>
  {()=>{
    const columns = ['name', 'age'];
    const data = [
      [
        'Jack',
        18,
      ],
      [
        'Amy',
        20,
      ],
      [
        'Mike',
        19,
      ],
      [
        'Lily',
        14,
      ],
    ];
    return (
      <div>
        <Table columns={columns} data={data} />
      </div>
    );
  }}
</Playground>

## Props

<Props of={Table} />
