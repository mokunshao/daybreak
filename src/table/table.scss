@import '../common.scss';

.daybreak-table {
  border: 1px solid $grey2;
  border-collapse: collapse;

  &-thead {
    border-bottom: 1px solid $grey2;
  }

  &-th {
    padding: $padding;
    text-align: left;

    &:not(:last-child) {
      border-right: 1px solid $grey2;
    }
  }

  &-td {
    padding: $padding;
  }
}
