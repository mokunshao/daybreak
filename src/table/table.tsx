import React, { TableHTMLAttributes, ReactNode } from 'react';
import { classes } from '../utils/classes';
import { joinedClass } from '../utils/joinedClass';
import './table.scss';

const table = joinedClass('table');

interface Props extends TableHTMLAttributes<HTMLTableElement> {
  columns: Array<string>;
  data: Array<Array<ReactNode>>;
}

export const Table: React.FC<Props> = (props) => {
  const {
    data, columns, className, ...rest
  } = props;
  return (
    <table className={classes(table(), className)} {...rest}>
      <thead className={table('thead')}>
        <tr>
          {columns.map((item) => <th className={table('th')} key={item}>{item}</th>)}
        </tr>
      </thead>
      <tbody>
        {data.map((row, index) => (
          <tr key={index}>
            {row.map((item, index2) => <td className={table('td')} key={index2}>{item}</td>)}
          </tr>
        ))}
      </tbody>
    </table>
  );
};

export default Table;
