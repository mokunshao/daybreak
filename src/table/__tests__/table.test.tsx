import React from 'react';
import { render } from '@testing-library/react';
import { Table } from '../table';

function Example() {
  const columns = ['name', 'age'];
  const data = [
    [
      'Jack',
      18,
    ],
    [
      'Amy',
      20,
    ],
    [
      'Mike',
      19,
    ],
    [
      'Lily',
      14,
    ],
  ];
  return (
    <div>
      <Table columns={columns} data={data} />
    </div>
  );
}


describe('Table', () => {
  it('exits', () => {
    expect(Table).toBeTruthy();
    expect(<Example />).toBeTruthy();
  });

  it('match snapshot', () => {
    const { asFragment } = render(<Example />);
    expect(asFragment()).toMatchSnapshot();
  });
});
