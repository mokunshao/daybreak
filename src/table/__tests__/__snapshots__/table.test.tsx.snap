// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table match snapshot 1`] = `
<DocumentFragment>
  <div>
    <table
      class="daybreak-table"
    >
      <thead
        class="daybreak-table-thead"
      >
        <tr>
          <th
            class="daybreak-table-th"
          >
            name
          </th>
          <th
            class="daybreak-table-th"
          >
            age
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="daybreak-table-td"
          >
            Jack
          </td>
          <td
            class="daybreak-table-td"
          >
            18
          </td>
        </tr>
        <tr>
          <td
            class="daybreak-table-td"
          >
            Amy
          </td>
          <td
            class="daybreak-table-td"
          >
            20
          </td>
        </tr>
        <tr>
          <td
            class="daybreak-table-td"
          >
            Mike
          </td>
          <td
            class="daybreak-table-td"
          >
            19
          </td>
        </tr>
        <tr>
          <td
            class="daybreak-table-td"
          >
            Lily
          </td>
          <td
            class="daybreak-table-td"
          >
            14
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
